cfgov/filing_instruction_guide/models.py

```diff
@@ -2,142 +2,94 @@
-                        "numbering": ""
-                        if is_appendix
-                        else format("{}.{} ", i, j),
-                        "children": [
-                            {
-                                "title": section3.header,
-                                "body": section3.body,
-                                "id": format("section-{}.{}.{}", i, j, k),
-                                "numbering": ""
-                                if is_appendix
-                                else format("{}.{}.{} ", i, j, k),
-                            }
-                            for k, section3 in enumerate(
-                                subsection.report_sections_level_three.all().order_by(
-                                    "pk"
-                                )
-                            )
-                        ],
-                    }
-                    for j, subsection in enumerate(
-                        section.report_subsections.all().order_by("pk")
-                    )
-                ],
-            }
-            for i, section in enumerate(sections)
-        ]
-
-    return sec
-
-
-get_report_sections = get_report_parts()
-get_report_appendices = get_report_parts(True)
-
-
-class ReportSection(ClusterableModel):
-    section_title = models.CharField(max_length=200, blank=True)
-    is_appendix = models.BooleanField(default=False)
-    section_content = StreamField(
-        [("full_width_text", organisms.FullWidthText())], blank=True
-    )
-    panels = [
-        FieldPanel("section_title"),
-        FieldPanel("is_appendix"),
-        StreamFieldPanel("section_content"),
-        InlinePanel("report_subsections", label="Subsection"),
-    ]
-    report = ParentalKey(
-        "FIGContentPage",
-        on_delete=models.CASCADE,
-        related_name="report_sections",
-    )
-
-
-class ReportSubSection(ClusterableModel):
-    sub_section_title = models.CharField(max_length=200)
-    sub_section_content = StreamField(
-        [("full_width_text", organisms.FullWidthText())], blank=True
-    )
-    panels = [
-        FieldPanel("sub_section_title"),
-        StreamFieldPanel("sub_section_content"),
-        InlinePanel(
-            "report_sections_level_three", label="Section Level Three"
-        ),
-    ]
-    section = ParentalKey(
-        "ReportSection",
-        on_delete=models.CASCADE,
-        related_name="report_subsections",
-    )
-
-
-class ReportSectionLevelThree(models.Model):
-    header = models.CharField(max_length=200)
-    body = models.TextField(blank=True)
-    subsection = ParentalKey(
-        "ReportSubSection",
-        on_delete=models.CASCADE,
-        related_name="report_sections_level_three",
-    )
-
-
+# function passed to context processor 
+def parse_section_builder(request, self):
+
+    # creates section dictionary object
+    def create_section_object(section, type_of_block):
```

Comment:
I think we can re-work the code to avoid having to do this, breaking up a wagtail block into its component parts and putting them into a new data structure. This does get the contents onto the page correctly, but I think there's a more straightforward method that uses built-in Wagtail methods and reduces the chance for error.